staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -347,6 +347,7 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	}
 	// we're not writing to this stream
 	errorStream.Close()
+	defer pf.streamConn.RemoveStreams(errorStream)
```

Comment:
I wonder what happens if the sreamConn is closed and we try to remove the streams after it is closed, see line405

```
	// always expect something on errorChan (it may be nil)
	err = <-errorChan
	if err != nil {
		runtime.HandleError(err)
		pf.streamConn.Close()
	}
```

Reply:
ah, it seems is ok

https://github.com/kubernetes/kubernetes/blob/1df24569a0bf62a528c49f73fdb236fd56eb05ee/staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/connection.go#L107-L132

Close already deletes the streams too